src/test/cypress/integration/exam/ExamParticipation.spec.ts

```diff
@@ -0,0 +1,122 @@
+                .build();
+            courseRequests.createExam(examContent).then((examResponse) => {
+                exam = examResponse.body;
+                courseRequests.registerStudentForExam(exam, student);
+                courseRequests.addExerciseGroupForExam(exam).then((groupResponse) => {
+                    courseRequests.createAndAddTextExerciseToExam(groupResponse.body, textExerciseTitle);
+                });
+                courseRequests.addExerciseGroupForExam(exam).then((groupResponse) => {
+                    courseRequests.createProgrammingExercise({ exerciseGroup: groupResponse.body });
+                });
+                courseRequests.generateMissingIndividualExams(exam);
+                courseRequests.prepareExerciseStartForExam(exam);
+            });
+        });
+    });
+
+    it('Participates as a student in a registered exam', () => {
+        startParticipation();
+        openTextExercise();
+        makeTextExerciseSubmission();
+        makeProgrammingExerciseSubmission();
+        handInEarly();
+        verifyFinalPage();
+    });
+
+    function startParticipation() {
+        cy.login(student, '/');
+        courses.openCourse(course.title);
+        courseOverview.openExamsTab();
+        courseOverview.openExam(examTitle);
+        cy.url().should('contain', `/exams/${exam.id}`);
+        examStartEnd.startExam();
+    }
+
+    function openTextExercise() {
+        examNavigation.openExerciseAtIndex(0);
+    }
+
+    function makeTextExerciseSubmission() {
+        const textEditor = artemis.pageobjects.textEditor;
+        textEditor.typeSubmission(submissionText);
+        // Loading the content of the existing files might take some time so we wait for the return of the request here
+        cy.intercept(GET, BASE_API + 'repository/*/files').as('getFiles');
+        textEditor.saveAndContinue().its('request.body.text').should('eq', submissionText);
+        cy.wait('@getFiles').its('response.statusCode').should('eq', 200);
+    }
+
+    function makeProgrammingExerciseSubmission() {
+        onlineEditor.createFileInRootPackage('placeholderFile');
+        onlineEditor.deleteFile('Client.java');
+        onlineEditor.deleteFile('BubbleSort.java');
+        onlineEditor.deleteFile('MergeSort.java');
+        onlineEditor.typeSubmission(submission, 'de.test');
+        onlineEditor.submit();
+        onlineEditor.getResultPanel().contains('100%').should('be.visible');
+        onlineEditor.getResultPanel().contains('13 of 13 passed').should('be.visible');
+        onlineEditor.getBuildOutput().contains('No build results available').should('be.visible');
+        onlineEditor.getInstructionSymbols().each(($el) => {
+            cy.wrap($el).find('[data-icon="check"]').should('be.visible');
+        });
+    }
+
+    function handInEarly() {
+        examNavigation.handInEarly();
+        examStartEnd.finishExam().its('response.statusCode').should('eq', 200);
```

Comment:
I think im guilty of doing this too but this assertion is essentially redundant because the test fails on a non 2xx response code anyways. Unless you wanted to check that the code is exactly 200.

Reply:
I think in this case cypress does not apply the 2xx or 3xx validation because it is not a request coming from cypress. Intercepted requests don't have that validation as far as I know